libtransmission/peer-mgr.cc

```diff
@@ -1353,16 +1353,8 @@ void create_bit_torrent_peer(
         return false;
     }
 
-    auto client = tr_interned_string{};
-    if (result.peer_id)
-    {
-        auto buf = std::array<char, 128>{};
-        tr_clientForId(std::data(buf), sizeof(buf), *result.peer_id);
-        client = tr_interned_string{ tr_quark_new(std::data(buf)) };
-    }
-
     result.io->set_bandwidth(&swarm->tor->bandwidth());
-    create_bit_torrent_peer(*swarm->tor, result.io, std::move(info), client);
+    create_bit_torrent_peer(*swarm->tor, result.io, std::move(info), result.peer_id);
```

Comment:
```suggestion
    create_bit_torrent_peer(*swarm->tor, result.io, std::move(info), result.peer_id.value_or(tr_peer_id_t{}));
```

Every peer is supposed to have a peer ID, so it makes more sense for `create_bit_torrent_peer()` and the constructors to accept `tr_peer_id_t` instead of `std::optional<tr_peer_id_t>`.

Reply:
OK, I think this is good enough, but just a bit of note to self for a potention future PR:

Tbh I think the `if` condition is not needed, and `tr_handshake::Result::peer_id` can be changed from an optional to plain `tr_peer_id_t` as well.

The `peer_id` optional will always contain a value when the program execution reaches where the `tr_clientForId()` function call was. The `if` condition was just for good measure, and the body is never jumped over in practice.